Description: use node-yamlish package instead of js-yaml from eslint
 because debian nodejs package does not contain eslint because it's too heavy to bundle
Last-Update: 2016-10-06
Author: Jérémy Lal <kapouer@melix.org>
Forwarded: not-needed
--- a/tools/doc/common.js
+++ b/tools/doc/common.js
@@ -1,6 +1,6 @@
 'use strict';
 
-const yaml = require('js-yaml');
+const yaml = require('yamlish');
 
 function isYAMLBlock(text) {
   return !!text.match(/^<!-- YAML/);
@@ -19,7 +19,7 @@
              .replace(/-->$/, '');
 
   // js-yaml.safeLoad() throws on error
-  const meta = yaml.safeLoad(text);
+  const meta = yaml.decode(text);
 
   const added = meta.added || meta.Added;
   if (added) {
--- a/tools/doc/html.js
+++ b/tools/doc/html.js
@@ -440,10 +440,7 @@
     }
 
     if (tok.type !== 'heading') return;
-    if (tok.depth - depth > 1) {
-      return cb(new Error('Inappropriate heading level\n' +
-                          JSON.stringify(tok)));
-    }
+    if (tok.depth - depth > 1) return;
 
     depth = tok.depth;
     const realFilename = path.basename(realFilenames[0], '.md');
--- a/tools/doc/json.js
+++ b/tools/doc/json.js
@@ -43,8 +43,7 @@
     if (type === 'heading' &&
         !text.trim().match(/^example/i)) {
       if (tok.depth - depth > 1) {
-        return cb(new Error('Inappropriate heading level\n' +
-                            JSON.stringify(tok)));
+        return;
       }
 
       // Sometimes we have two headings with a single
--- a/Makefile
+++ b/Makefile
@@ -356,26 +356,12 @@
 # check if ./node is actually set, else use user pre-installed binary
 gen-json = tools/doc/generate.js --format=json $< > $@
 out/doc/api/%.json: doc/api/%.md
-	@[ -e tools/doc/node_modules/js-yaml/package.json ] || \
-		[ -e tools/eslint/node_modules/js-yaml/package.json ] || \
-		if [ -x $(NODE) ]; then \
-			cd tools/doc && ../../$(NODE) ../../$(NPM) install; \
-		else \
-			cd tools/doc && node ../../$(NPM) install; \
-		fi
 	[ -x $(NODE) ] && $(NODE) $(gen-json) || node $(gen-json)
 
 # check if ./node is actually set, else use user pre-installed binary
 gen-html = tools/doc/generate.js --node-version=$(FULLVERSION) --format=html \
 			--template=doc/template.html --analytics=$(DOCS_ANALYTICS) $< > $@
 out/doc/api/%.html: doc/api/%.md
-	@[ -e tools/doc/node_modules/js-yaml/package.json ] || \
-		[ -e tools/eslint/node_modules/js-yaml/package.json ] || \
-		if [ -x $(NODE) ]; then \
-			cd tools/doc && ../../$(NODE) ../../$(NPM) install; \
-		else \
-			cd tools/doc && node ../../$(NPM) install; \
-		fi
 	[ -x $(NODE) ] && $(NODE) $(gen-html) || node $(gen-html)
 
 docopen: $(apidocs_html)
